Fix sign of phase shift in DataSet.apply_symop

JBGreisman opened the pull request:
Based on visual inspection of maps, it appeared that `DataSet.apply_symop()` was applying phase shifts in the opposite direction of the specified symmetry operation. This PR changes the sign for how phase shifts are applied, and updates the relevant test.

JBGreisman commented:
To do:
 - It would be useful to make a "gold standard" pair of F/phi with a known real-space translation between the corresponding maps so that we can write a cleaner test.

JBGreisman commented:
My most recent push added a test that uses fmodel data to compute a real-space map, shifts the map based on a desire translation using `np.roll`, back computes corresponding complex structure factors. These are then compared to the results of `DataSet.apply_symop()`

kmdalton commented on reciprocalspaceship/dataset.py:
```diff
@@ -597,8 +597,7 @@ def apply_symop(self, symop, inplace=False):
+            dataset[key] = phic * (dataset[key] - np.rad2deg(phase_shifts))
             dataset[key] = utils.canonicalize_phases(dataset[key], deg=True)
         for key in dataset.get_complex_keys():
             dataset[key] *= np.exp(1j*phase_shifts)
```
Don't you need to change the sign of this complex rotation too?

JBGreisman replied:
Fixed in latest update -- the phase shifts have now been flipped for complex structure factors as well

kmdalton commented:
As per our conversation offline, let's go ahead and merge this and then do a refactor.

kmdalton reviewed (changes requested):
I am not clear why you did not change the sign of the phase shift applied to complex structure factors. Not sure if this is in error. If tests are passing, I am concerned that complex phase shifts are maybe not adequately tested. 

Can you either change the sign of complex phase shifts or tell me that there is no need for some reason that I am missing?

JBGreisman commented:
I am still working on this for the complex structure factors. This test may not be working as well as I had hoped, but I am not yet ready to come to that conclusion

JBGreisman commented:
This PR now fixes `DataSet.apply_symop()` for use with both |F|+phi and complex SFs. I updated the test to use a P1 dataset avoid an interdependency caused by the use of `apply_symop()` in `DataSet.to_reciprocalgrid()`. 

P.S. -- I updated my emacs recently to auto-apply `black` formatting, which is why there are additional changes to the `test_dataset_symops.py` file

kmdalton commented on tests/test_dataset.py:
```diff
@@ -495,14 +495,14 @@ def test_apply_symop_roundtrip(mtz_by_spacegroup):
     dataset = rs.read_mtz(mtz_by_spacegroup)
     for op in dataset.spacegroup.operations():
         applied = dataset.apply_symop(op)
-        back = applied.hkl_to_asu()
```
We should try to resurrect this test when we do the refactor, but this is okay with me for now.

kmdalton reviewed (approved):
This looks good to me.
